l1.py

```diff
@@ -0,0 +1,395 @@
+import http.client
+import os
+import shutil
+import socket
+import subprocess
+import sys
+import time
+from os.path import join as pjoin
+
+import libroll as lib
+# from l1_genesis import GENESIS_TMPL
+from paths import OPPaths
+from processes import PROCESS_MGR
+
+sys.path.append("optimism/bedrock-devnet/devnet")
+# noinspection PyUnresolvedReferences
+from genesis import GENESIS_TMPL
+
+
+####################################################################################################
+
+DEVNET_L1_DATA_DIR = "db/devnetL1"
+"""Directory to store the devnet L1 blockchain data."""
+
+
+####################################################################################################
+
+def deploy_l1_devnet(paths: OPPaths):
+    """
+    Spin the devnet L1 node, doing whatever tasks are necessary, including installing geth,
+    generating the genesis file and config files, and deploying the L1 contracts.
+    """
+    os.makedirs(paths.devnet_l1_gen_dir, exist_ok=True)
+
+    patch(paths)
+    generate_devnet_l1_genesis(paths)
+    start_devnet_l1_node(paths)
+    generate_devnet_l1_config(paths)
```

Comment:
can generate config before start L1 node? I think it should be better if we can do provision work before start the service, it will also benefit when we support different deployment platform besides localhost.

Reply:
The "devnet L1 config" is a actually a "devnet" or "network" config and is not used for spinning the L1. I've changed the name to clarify, and changed the order too (but doesn't matter)

Reply:
https://github.com/0xFableOrg/simple-op-stack-rollup/issues/24